lib/Conversion/GPUXToLLVM/GPUXToLLVMPass.cpp

```diff
@@ -0,0 +1,645 @@
+//===- GPUToSPIRVPass.cpp -  --------------*- C++ -*-===//
+//
+// Copyright 2022 Intel Corporation
+// Part of the IMEX Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file extends upstream GPUToSPIRV Pass that converts GPU ops to SPIR-V
+/// by adding more conversion patterns like SCF, math and control flow. This
+/// pass only converts gpu.func ops inside gpu.module op.
+///
+//===----------------------------------------------------------------------===//
+#include "imex/Conversion/GPUXToLLVM/GPUXToLLVMPass.h"
+#include "imex/Dialect/GPUX/IR/GPUXOps.h"
+
+#include "imex/Transforms/FuncUtils.hpp"
+#include "imex/Transforms/TypeConversion.hpp"
+
+#include "../PassDetail.h"
+
+// TODO: remove
+#include <mlir/Dialect/Func/IR/FuncOps.h>
+#include <mlir/Dialect/Func/Transforms/FuncConversions.h>
+//
+
+#include "mlir/Conversion/GPUCommon/GPUCommonPass.h"
+
+#include "mlir/Conversion/ArithToLLVM/ArithToLLVM.h"
+#include "mlir/Conversion/AsyncToLLVM/AsyncToLLVM.h"
+#include "mlir/Conversion/ControlFlowToLLVM/ControlFlowToLLVM.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVMPass.h"
+#include "mlir/Conversion/LLVMCommon/ConversionTarget.h"
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Conversion/MemRefToLLVM/MemRefToLLVM.h"
+#include "mlir/Conversion/VectorToLLVM/ConvertVectorToLLVM.h"
+#include "mlir/Dialect/Async/IR/Async.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/GPU/Transforms/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FormatVariadic.h"
+
+#include <iostream>
+
+namespace imex {
+
+static constexpr const char *kGpuBinaryStorageSuffix = "_spirv_binary";
+
+struct FunctionCallBuilder {
```

Comment:
all internal classes must be inside unnamed namespace

Reply:
done.